utils/build-script-impl

```diff
@@ -641,7 +641,10 @@ if [[ "$SKIP_TEST_WATCHOS" ]] ; then
     SKIP_TEST_WATCHOS_SIMULATOR=1
 fi
 
-if [[ "${CMAKE_GENERATOR}" == "Ninja" ]] && [[ -z "$(which ninja)" ]] ; then
+# Some Linux distributions, e.g. Fedora, use 'ninja-build' instead of
+# 'ninja' as binary name. Check for both.
+if [[ "${CMAKE_GENERATOR}" == "Ninja" ]] && [[ -z "$(which ninja)" ]] &&
+   [[ -z "$(which ninja-build)" ]] ; then
```

Comment:
Wouldn't this fail overtime since either one of the two would yield false? (Not having ninja but having ninja-build and vice versa).